### ISSUE TITLE
Confusing the LightningModule for the Trainer

### ISSUE THREAD

**Issue by chiragraman** [gabinsane/multimodal-vae-comparison#1]
Hi @gabinsane !

Thanks for working on this, and making so many improvements so quickly, (especially incorporating Lightning, all your abstract base classes, and separating the code into modules). 

A quick point about the lightning module you have (`Trainer`). Lightning already provides its own trainer class that does the training (along with allowing for callbacks to be triggered at the end of each train step, train epoch etc). This is the `pl_trainer` in `main.py`. The LightningModule itself is supposed to be your overall system where one can find all relevant details: the model, the optimizers, the loss functions, what happens in a train step, val step etc. The lightning Trainer class then accepts the LightningModule and trains using some training parameters (including checkpointing callbacks).

I believe the way you have it setup right now confuses the LightningModule for the Trainer (e.g. in `main.py`). I believe the LightningModule should essentially be named `MultimodalVAE` or something along those lines.

**Comment by gabinsane** [comment]
Hi @chiragraman, thank you for all your recommendations on how to make this toolkit better. 
We are new to the Lightning module, so it might take a bit more time until everything is optimized.
I just pushed a commit renaming the Trainer class to MultimodalVAE, thank you for noticing!
